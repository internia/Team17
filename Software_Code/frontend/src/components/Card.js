import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components';

const Styles = styled.div`

    .checkBtn {
        text-align: right;
        background-color: #17A2B8;
        color: white;
    }

    .modal{
        display: none;
        position: fixed;
        z-index: 1;
        left: 0;
        top:0;
        width: 100%;
        height: 100%;
        overflow: auto;
        background-color: rgb(0,0,0);
        background-colour: (0,0,0,0.4);
    }

    .modal-content{
        background-color: #fefefe;
        margin: 15% auto;
        padding: 20px;
        border: 1px solid #888;
        width: 80%;
    }

    :hover {
        background-color: #273c75;
        color: white;
		cursor: pointer;

`;

export class Card extends Component {
    
    getStyle = () => {
        return {
            padding: '0.7em',
            fontSize: '0.7em',
        }
    }

    render() {
        return (
            <Styles> 
            <div style={{'margin-bottom': '1em', 'padding':'1em', backgroundColor: '1#dcdde1', border: '1px solid'}}>
             <div style={{fontSize: '1.4em', fontweight:600}} class="row row-cols-2">
              
              <div class="col"> <b>{this.props.card.ProviderName}</b></div>
              
            </div>

            <div class="row row-cols-2">
                <div class="col">{this.props.card.DRGDefinition}</div>
            </div>

            <div class="row row-cols-2">
                <div class="col"><b>GPDID Code: </b>{this.props.card.GPDID}</div>
            </div>
                
            <div class="row row-cols-2">
                <div class="col"><b>Average Price: </b></div>
                <div class="col">${this.props.card.TotalPayments}</div>
            </div>

            <div class="row row-cols-2">
                <div class="col"><b>Distance: </b></div>
                <div class="col">256 Miles</div>
            </div>

            <div class="row row-cols-2">
                <div class="col"><b>More information: </b></div>
                <div class="col">BUTTON</div>
            </div>

        </div>
     </Styles>

        )
    }
}

Card.propTypes = {
    card: PropTypes.object.isRequired
}

export default Card
