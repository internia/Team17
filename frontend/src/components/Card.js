import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components';


const Styles = styled.div`

    .checkBtn {
        position: relative;
        text-align: center;
        background-color: #17A2B8;
        color: white;
        padding: 10px;
        border: none;
        border-radius: 10px;
        margin-right: 30px;
        right: -800px;
        bottom: 40px;
    }

    .modal{
        display: none;
        position: fixed;
        z-index: 1;
        left: 0;
        top:0;
        width: 100%;
        height: 100%;
        overflow: auto;
        background-color: rgb(0,0,0);
        background-colour: (0,0,0,0.4);
    }

    .modal-content{
        background-color: #fefefe;
        margin: 15% auto;
        padding: 20px;
        border: 1px solid #888;
        width: 80%;
    }

`;

export class Card extends Component {
    
    getStyle = () => {
        return {
            padding: '0.7em',
            fontSize: '1em',
            backgroundColor: '#ccebdb'
        }
    }

    
    
    render() {
        return (
            <Styles> 
            <div style={{'margin-top': '1em'}} className={'row'}>
                <div className={'col-12'} style={this.getStyle()}><button style={{'margin-right': '1em'}} className={'btn-lg btn btn-info'}>{this.props.card.ID}</button> {this.props.card.DRGDefinition}
                <button className={'checkBtn'}>Check Price</button> <button className={'checkBtn'}> Check Hospitals</button> 
                 </div>
                {/* <div id="priceModal" class="modal">
                    <div class="priceModalContent">
                        <span class="close"> </span>
                        <p>weoowooow</p>
                    </div>
                </div>  */}
            
            </div>
            </Styles>
        )
    }
}

// var modal = document.getElementById("myModal");

// var checkBtn = document.getElementById("myBtn");

// var span = document.getElementsByClassName("close")[0];

// checkBtn.onclick = function(){
//     modal.style.display = "block";
// }

// span.onclick = function() {
//     modal.style.display = "none";
// }

// window.onclick = function(event){
//     if(event.target === modal){
//         modal.style.display = "none";
//     }
// }
Card.propTypes = {
    card: PropTypes.object.isRequired
}

export default Card
