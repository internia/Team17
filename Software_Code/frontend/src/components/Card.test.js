import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { isCompositeComponent } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() })


it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Card key={1} card={{
        ProviderName: 'test driven development',
        DRGDefinition: 'testDefinition',
        GPDID: "GDPID Test",
        TotalPayments: 10909,
    }} ></Card>, div)
})

it('renders searchbar correctly', () => {
    function callBack() {
        console.log("called back")
    }
    render(<Card card={{
        ProviderName: 'test',
        DRGDefinition: 'testDefinition',
        GPDID: "GDPID Test",
        TotalPayments: 10909,
    }} ></Card>)
})


it("matches snapshot", () => {
    function callBack() {
        console.log("called back")
    }
    const tree = renderer.create(<Card key={1} card={{
        ProviderName: 'test driven development',
        DRGDefinition: 'testDefinition',
        GPDID: "GDPID Test",
        TotalPayments: 10909,
    }} ></Card>)
    expect(tree).toMatchSnapshot();
    // expect(true).toBeTruthy();
})