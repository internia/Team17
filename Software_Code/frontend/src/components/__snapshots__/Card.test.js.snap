// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AykKC JXmus"
>
  <div
    style={
      Object {
        "backgroundColor": "1#dcdde1",
        "border": "1px solid",
        "margin-bottom": "1em",
        "padding": "1em",
      }
    }
  >
    <div
      class="row row-cols-2"
      style={
        Object {
          "fontSize": "1.4em",
          "fontweight": 600,
        }
      }
    >
      <div
        class="col"
      >
         
        <b>
          test driven development
        </b>
      </div>
    </div>
    <div
      class="row row-cols-2"
    >
      <div
        class="col"
      >
        testDefinition
      </div>
    </div>
    <div
      class="row row-cols-2"
    >
      <div
        class="col"
      >
        <b>
          GPDID Code: 
        </b>
        GDPID Test
      </div>
    </div>
    <div
      class="row row-cols-2"
    >
      <div
        class="col"
      >
        <b>
          Average Price: 
        </b>
      </div>
      <div
        class="col"
      >
        $
        10909
      </div>
    </div>
    <div
      class="row row-cols-2"
    >
      <div
        class="col"
      >
        <b>
          Distance: 
        </b>
      </div>
      <div
        class="col"
      />
    </div>
  </div>
</div>
`;
